### PR TITLE
Remove the secondary controller from turn()

### PR DESCRIPTION
**Pull request by bradamiller**
The secondary controller caused jittery turning. Removed it.

### CODE REVIEW

File: XRPLib/differential_drive.py [hunk 1]
```diff
@@ -220,70 +219,49 @@ def turn(self, turn_degrees: float, max_effort: float = 0.5, timeout: float = No
...
-        :param use_imu: A boolean flag that changes if the main controller bases its movement off of the imu (True) or the encoders (False)
-        :type use_imu: bool
+        :param controller: The main controller, for handling the angle turned
+        :type controller: Controller
         :return: if the distance was reached before the timeout
         :rtype: bool
         """
-
+        
+        turn_speed = max_effort * 90
```
Comment (fgrossman) [inline XRPLib/differential_drive.py:228]: I'm not sure I understand the need for this statement.

File: XRPLib/differential_drive.py [hunk 2]
```diff
@@ -220,70 +219,49 @@ def turn(self, turn_degrees: float, max_effort: float = 0.5, timeout: float = No
...
+        
         if max_effort < 0:
-            max_effort = -max_effort
             turn_degrees = -turn_degrees
```
Comment (fgrossman) [inline XRPLib/differential_drive.py:231]: It feels like there should be a check for -degrees. What if I request -90 degrees at -1 speed?

File: XRPLib/encoded_motor.py
Comment (fgrossman) [inline XRPLib/encoded_motor.py:1]: Are the changes here relative to the PR for turn()?

File: XRPLib/differential_drive.py (hunk 1 above)
Comment (bradamiller) [reply XRPLib/differential_drive.py:228]: This translates the robot turn rate into a wheel turn speed speed. In the previous version there was a secondary PID controller and that took the robot turn rate, but now we just set the wheel speeds which effectively sets the turn speed.

**Comment by AnselChang** [comment]
Closed in favor of https://github.com/Open-STEM/XRP_MicroPython/pull/67

**Pull request closed without merging**